src/libs/ReportUtils.ts

```diff
@@ -11816,6 +11816,22 @@ function getReportStatusTranslation(stateNum?: number, statusNum?: number): stri
     return '';
 }
 
+/**
+ * Checks if a workspace member is leaving a workspace room
+ * This is used to determine if we need to show special handling when a workspace member leaves a room
+ */
+function isWorkspaceMemberLeavingWorkspaceRoom(report: OnyxEntry<Report>, isPolicyEmployee: boolean, isPolicyAdminParam: boolean): boolean {
+    if (!report) {
+        return false;
+    }
+    return (
+        (report.visibility === CONST.REPORT.VISIBILITY.RESTRICTED ||
+            // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+            (isPolicyExpenseChat(report) && (report.isOwnPolicyExpenseChat || isPolicyAdminParam))) &&
```

Comment:
This feels confusing to read, can we break this apart into a couple of variables to make it clearer?